src/Tools/CoreFx.Tools/IsManagedAssembly.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.DotNet.Build.Tasks
+{ 
+    public partial class GenerateDepsJson
+    {
+        public bool IsManagedAssembly(string file, out AssemblyName result)
+        {
+            result = null;
+            try
+            {
+                var stream = File.ReadAllBytes(file).ToImmutableArray();
```

Comment:
I think I did so that it would build for both core and desktop. I was having some trouble getting the PEReader api to work.